src/gui/issueswidget.cpp

```diff
@@ -50,6 +50,59 @@ bool persistsUntilLocalDiscovery(const OCC::ProtocolItem &data)
 }
 namespace OCC {
 
+class SyncFileItemStatusSetSortFilterProxyModel : public QSortFilterProxyModel
+{
+public:
+    SyncFileItemStatusSetSortFilterProxyModel(QObject *parent = nullptr)
+        : QSortFilterProxyModel(parent)
+    {
+    }
+
+    QSet<SyncFileItem::Status> filter() const
+    {
+        return _filter;
+    }
+
+    void setFilter(const QSet<SyncFileItem::Status> &newFilter)
+    {
+        if (_filter != newFilter) {
+            _filter = newFilter;
+            invalidateFilter();
+        }
+    }
+
+    bool filterAcceptsRow(int sourceRow, const QModelIndex &sourceParent) const
+    {
+        if (_filter.isEmpty()) {
+            return true;
+        }
+
+        QModelIndex idx = sourceModel()->index(sourceRow, filterKeyColumn(), sourceParent);
+        return _filter.contains(static_cast<SyncFileItem::Status>(data(idx, filterRole()).toInt()));
+    }
+
+private:
+    QSet<SyncFileItem::Status> _filter;
```

Comment:
Could you see an issue in converting the enum to QFlags? It would simplify the filter by a lot.

Reply:
A comment at the enum states that it fits in 4 bits, so I thought that turning that into flags would break something...

Reply:
Ah yes the 4bit optimisation ......

Reply:
So should I change it or not?